B5500: Avoid potential truncataion warning with type consistency

markpizz opened the pull request:
Some compilers correctly generate this warning:

simh-markpizz\B5500\b5500_sys.c(349,13): warning C4244: '=': conversion from 't_value' to 'uint
16', possible loss of data [C:\Users\Mark\Documents\simh-markpizz\BIN\cmake-vs2019\b5500.vcxproj]

This is one way to fix this.  A direct cast would be fewer changes, but all the input data is already uint16 or even 12 bits, so this seemed appropriate.

OK to merge.

BTW, there was a minor change to sim_card.c recently which you should migrate back to your repo...

rcornwell commented:
I will have to verify these changes don't break anything.

I have pulled in the latest SCP as of today. I pulled the sim_card changes in the other day.

markpizz commented:
I looked at this carefully.  print_opcode is called in 3 places:
  1)  in b5500_cpu.c with an explicit uint16 argument from the saved history record.
  2) in b5500_sys.c/344 with what was an int argument that is computed with a shift and then a masked with 07777.  I changed the int to uint16.
  3) in b5500_sys.c/413 with what was an int argument that is computed with a shift and then a masked with 07777.  I changed the int to uint16.

I merely created this a pull request to show you the code and give you the choice of either doing it this way or merely casting the t_value to uint16 inside of print_opcode.  Since all the callers are really dealing with 12 or 16 bits of data, it seemed appropriate to change the argument type from t_value to uint16.

markpizz commented:
I'm closing this pull request now since someone else has also generated another pull request and the change history won't be sequential.

Please let me know if the change I suggested was OK and I'll merge the commit behind other changes which are happening.

rcornwell commented:
This looks ok, I will apply it to my repo and push it over to the main simh tonight.
